modules/decision_reviews/app/controllers/decision_reviews/v1/supplemental_claims_controller.rb

```diff
@@ -141,6 +144,55 @@ def clear_in_progress_form
       def error_class(method:, exception_class:)
         "#{self.class.name}##{method} exception #{exception_class} (SC_V1)"
       end
+
+      # To conform to the LH schema, we need to ensure that if the evidenceType includes 'retrieval',
+      # then the retrieveFrom array must have facilities with unique facility location names
+      def normalize_evidence_retrieval_for_lighthouse_schema(req_body_obj)
+        evidence_submission = req_body_obj.dig('data', 'attributes', 'evidenceSubmission')
+        evidence_type = evidence_submission&.dig('evidenceType')
+        retrieve_from = evidence_submission&.dig('retrieveFrom')
+        # Return if evidenceType is nil or doesn't include 'retrieval'
+        return req_body_obj unless evidence_type.is_a?(Array) && evidence_type.include?('retrieval')
+        # Return if retrieveFrom is nil, not an array, or only one item
+        return req_body_obj unless retrieve_from.is_a?(Array) && retrieve_from.length > 1
+
+        grouped_entries = retrieve_from.group_by do |entry|
+          entry.dig('attributes', 'locationAndName')
+        end
+
+        merged_entries = grouped_entries.map do |_, entries|
+          if entries.length == 1
+            entries.first
+          else
+            merge_evidence_entries(entries)
+          end
+        end
+
+        req_body_obj['data']['attributes']['evidenceSubmission']['retrieveFrom'] = merged_entries
+        req_body_obj
+      end
+
+      def merge_evidence_entries(entries)
+        merged_entry = entries.first.deep_dup
+        merged_attributes = merged_entry['attributes']
+
+        all_evidence_dates = []
+        entries.each do |entry|
+          attributes = entry['attributes']
+          if attributes && attributes['evidenceDates'].is_a?(Array)
+            all_evidence_dates.concat(attributes['evidenceDates'])
+          end
+        end
+
+        unique_evidence_dates = all_evidence_dates.uniq.sort_by { |date_range| date_range['startDate'] }
+
+        # Apply schema constraints (maxItems: 4) -- this should be unlikely,
+        # as we should only be collecting dates for pre-2005 treatment
+        unique_evidence_dates = unique_evidence_dates.first(4)
+
+        merged_attributes['evidenceDates'] = unique_evidence_dates
+        merged_entry
+      end
```

Comment:
Again, I think these could be broken into multiple, smaller methods. Not a blocker, but something to consider next time.